Migrate to go1.12 to simplify our cmd/http package

harshavardhana opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Migrate to go1.12 to simplify our cmd/http package
<!--- Describe your changes in detail -->

## Motivation and Context
Simplify the cmd/http package overall by removing
custom plain text v/s tls connection detection, by
migrating to go1.12 and choose a minimum version
to be go1.12

Also remove all the vendored deps, since they
are not useful anymore.
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## Regression
No
<!-- Is this PR fixing a regression? (Yes / No) -->
<!-- If Yes, optionally please include minio version or commit id or PR# that caused this regression, if you have these details. -->

## How Has This Been Tested?
Using go 1.12
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

wlan0 commented on Dockerfile.simpleci:
```diff
@@ -24,17 +25,14 @@ RUN echo 'Defaults    secure_path="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr
 RUN for d in $(go list ./... | grep -v browser); do go test -v -race --timeout 15m "$d"; done
 RUN make verifiers
 RUN make crosscompile
-RUN make coverage || true
```
we just need to remove the catch all fail-safe

harshavardhana replied:
yeah, I removed coverage because it was useful previously to report this value to codecov.io - through SimpleCI if we need to report we need to add one more tool to report to CI the code-coverage.

wlan0 replied:
gotcha

harshavardhana committed:
Migrate to go1.12 to simplify our cmd/http package

Simplify the cmd/http package overall by removing
custom plain text v/s tls connection detection, by
migrating to go1.12 and choose minimum version
to be go1.12

Also remove all the vendored deps, since they
are not useful anymore.
